splash/network_manager.py

```diff
@@ -226,7 +240,12 @@ def _handle_custom_headers(self, request):
             headers = headers.items()
 
         for name, value in headers or []:
-            request.setRawHeader(to_bytes(name), to_bytes(value))
+            try:
+                request.setRawHeader(to_bytes(name), to_bytes(value))
+            except TypeError:
+                msg = "invalid header {}: {}. Header keys and values must be string or bytes"
+                self.log(msg.format(name, value), min_level=1, format_msg=False)
+                continue
```

Comment:
Yeah, but these values are usually set in splash:on_request, and it is hard to get an error from there, so it is harder for user to detect this error in script. Errors are raised so that users can fix their scripts and make them more robust, but here error doesn't serve this need because it is not obvious how to fix a script after this error (it is only in the logs). 

You're right that we should make it robust (and so we may have to catch str() errors), but I think it worths it; the alternative is worse.